### ISSUE TITLE
Add keyboard DOM widget

### ISSUE THREAD

**Issue by benbovy** [benbovy/ipytone#10]
An example here: https://codepen.io/andrewphillipdoss/pen/rJXeeW

**Comment by benbovy** [comment]
NexusUI provides a keyboard widget too: http://nexus-js.github.io/ui/api/#Piano

**Comment by davidbrochart** [comment]
We could also interface with external MIDI devices using the set of Web MIDI APIs, e.g. https://developer.mozilla.org/en-US/docs/Web/API/MIDIInput.

**Comment by benbovy** [comment]
Yes that would be great. Similarly to the Web Audio API -> Tones.js, there's [WebMIDI](https://github.com/djipco/webmidi). There's even which has already been used [here](https://github.com/jupyter-widgets/midicontrols).

**Comment by benbovy** [comment]
As an alternative to MIDI, we could also look to libraries implementing the [OSC](https://fr.m.wikipedia.org/wiki/Open_Sound_Control) protocol (Python and/or JS).

**Comment by benbovy** [comment]
[react-piano](https://www.npmjs.com/package/react-piano) looks very nice, unless we don't want react as a dependency.
